src/docs/frontend/component-library.mdx

```diff
@@ -0,0 +1,151 @@
+- If you are testing a hook then you must pass a string as the first argument to `storyBook()` instead of the hook itself because.
+
+### Non-framework stories
+
+It is possible to skip the `storyBook()` framework function and build stories from scratch. 
+
+Doing this allows full flexibility to define your story.
+
+Using default exports, named exports, or a mixture of each is fully supported. But take note that the order they will be rendered is not consistent/defined.
+
+Look at [icons.stories.tsx](https://sentry.io/orgredirect/organizations/:orgslug/stories/?name=app/icons/icons.stories.tsx) for an example.
+
+### Helper Components
+
+There are some helper components specifically built to make common patterns in stories easier to implement. Of course you can import anything in the sentry repo to make your stories richer, and create new helpers to make expressing stories easier!
+
+Helper Components:
+
+<IconFire color="gray500" size="sm" />
+
+- [`<JSXNode />`](https://github.com/getsentry/sentry/blob/master/static/app/components/stories/jsxNode.tsx)<br />
+    Render a formatted JSX component name and some properties.<br />
+    ```typescript
+    return <JSXNode name="IconFire" props={{color: 'red400', size: 'sm'}} />;
+
+    // will render as <code>&lt;IconFire<br/>color="red"<br/>size="sm" \:gt;</code>
+    ```
+
+- [`<JSXProperty />`](https://github.com/getsentry/sentry/blob/master/static/app/components/stories/jsxProperty.tsx)
+    Render a formatted JSX property name & value.<br />
+    ```typescript
+    return <JSXProperty name="disabled" value={false} />;
+
+    // will render as `<code>size={false}</code>`
+    ```
+
+- [`<SideBySide>{children}</SideBySide>`](https://github.com/getsentry/sentry/blob/master/static/app/components/stories/sideBySide.tsx)<br />
+    A shortcut for `display: flex;` to render multiple items in a row, wrapping as needed
+    ```typescript
+    return (
+      <SideBySide>
+        <Badge type="default">Default</Badge>
+        <Badge type="alpha">Alpha</Badge>
+        <Badge type="beta">Beta</Badge>
+      </SideBySide>
+    );
+    ```
+
+- [`<SizingWindow />`](https://github.com/getsentry/sentry/blob/master/static/app/components/stories/sizingWindow.tsx)<br />
+    A wrapper component to help demonstrate what the component looks like when the parent size is different or changing.<br />
+    By default uses `display:flex; overflow: hidden;` which can test responsive components well. Can also be set to `display: block; overflow: auto;`.<br />
+    ```typescript
+    <SizingWindow style={{height: '100px'}}>
+      <LoadingTriangle />
+    </SizingWindow>`
+    ```
+
+- [`<Matrix />`](https://github.com/getsentry/sentry/blob/master/static/app/components/stories/matrix.tsx)<br />
+    A helper to render multiple pairs of properties and values in a grid.
+    For example, we can render all button `priority` values against all possible `size` values which results in a 4x5 matrix of outputs.
```

Comment:
might be more clear if you say something like "render 2 button `priority` values against 2 possible `size` values which results in a 2x2..." which matches the example below, or you could match the example to the text

Reply:
```suggestion
    For example, we can render N values for one props, against M values for another prop. The example below compares 4 values for button `size` against 5 values for `priority`, resulting in a 20 example buttons in a grid.
```